REVIEW.md

```diff
@@ -0,0 +1,219 @@
+Reviewers should recall that code review is primarily about reducing risk in the project.
+When reviewing code, one should aim at preventing situations which may require a bug fix, a deprecation, or a retraction.
+
+- [ ] Improvements
+    - Could the code easily be rewritten to run much more efficiently for relevant settings?
+    - Is the code backwards compatible with previous versions?
+      (Or is a deprecation cycle necessary?)
+    - Is there feedback that would be helpful for solving future issues surrounding this change?
+      Are there non-urgent concerns that the team should be made aware of?
+    - If a bug is fixed, does the PR include a non-regression test?
+    - Are there images that should be added?
+      Are there images that should be resized or optimized in another way?
+
+- [ ] User-facing changes
+    - Does the user interface or other visual elements match existing patterns in Spyder?
+      If it does not, is there a reason?
+
+Review that addresses these questions on a PR will be prefixed with `[optional]` so that the PR author(s) are clear that this feedback is not a blocker.
+Addressing this feedback is considered above-and-beyond, and may be more relevant on some PRs than others.
+PR authors may address these reviews, but are not required to do so in order for the PR to be eligible for merging.
+
+
+## For documentation-focused review
+
+### Required review
+
+The following are areas of review the community expects to be covered in the review of a documentation-focused (sometimes abbreviated as docs) PR.
+In general, reviewers need to consider whether the content is a good fit, accurate and generally clear to readers, and builds and renders with no simmediate user-visible issues.
+
+**Motivation**
+
+- [ ] Does the change improve the documentation?
+- [ ] Would it help multiple users to have this as a part of the documentation?
+- [ ] Is it appropriate for the documentation and the section it is placed in?
+
+**Content**
+
+- [ ] Is the information accurate?
+- [ ] Is the change complete, such that it won’t confuse users if it were to appear in the documentation as-is?
+- [ ] Is it clear what the text and images are trying to communicate?
+
+**Images/GIFs**
+
+- [ ] Are screenshots/GIFs reasonably legible and appropriate for the topic?
+- [ ] Are all images/GIFs scaled and optimized?
+If needed, reviewers should offer to help with this.
+
+**Technical**
+
+- [ ] Do the tests pass in the continuous integration build?
+If appropriate, help the contributor understand why tests failed.
+- [ ] Does any added/changed content render correctly, without user-visible issues?
+- [ ] Is the content free from typos and unambiguous grammar errors (review the [Style Guide](https://github.com/spyder-ide/spyder-docs/blob/master/STYLEGUIDE.md) for more information)?
+
+Review that addresses these questions on a PR will be prefixed with `[required]` so that the PR author(s) are clear that they are blockers.
```

Comment:
I very deliberately chose not to include line this in this section (and would recommend reframing or removing it in the code-focused review section as well), for a number of reasons.

Most importantly, I feel having to tag (almost every, at least in most cases on the docs going forward) review comment/suggestion as `[required]` is not friendly very toward contributors, in several ways. Labeling suggestions as "required" carries with it the implicit presumption that the reviewer's judgement is inherently correct, the contributor is not and they _must_ make the change. This is certainly not always the case even on new contributor PRs, and might dissuade them or intimidate them against questioning it, even if they might think they have a clear reason (and end up being right).

Furthermore, it may not be appropriate or send the desired message for a large fraction of comments and in fact make reviews sound much more harsh than reviewers intend them to, particularly in the case of the large fraction of comments where the reviewer is varying degrees of unsure about the change they are proposing and wants to make that, and that they value the contributor's feedback, clear. Or, they may even want to frame it as a nuetral question to begin with—a much softer and more inclusive approach, that a prominent `[required]` tag can detract from. All told, comments that fall into one of these categories can sometimes comprise nearly half of all the comments on many of those I've been involved in lately. See [my comment above](https://github.com/spyder-ide/spyder/pull/16789#discussion_r747120076) for more background on that.

In addition to that, a required `[required]` tag has several other downsides, without much clear advantage. Given the large to overwhelming majority of allowed comments would include it, it makes for a greater learning curve, tedious work, and more mental energy for the reviewer to add to each one. Furthermore, it increases the verbosity, "noiseness" and complexity of the review for the contributor, requiring more time and energy to read. Finally, perhaps most importantly, just tagging the `[optional]` review comments makes it much easier for both reviewers and authors to spot at a glance whether a certain comment is optional or not, given both tags look visually similar and its much quicker to scan for tag/no tag than to parse the specific word in the tag, and also draws more attention to the fact that optional comments are optional as opposed to required comments are required, creating a more postitive overall impression.

Reply:
I agree with @cam-gerlach's sentiments. It will become quite tedious to require `[required]` or `[optional]` tags for most or all comments. It is my understanding that PRs generally are blocked until all review conversations are marked as resolved; perhaps that is not correct. Does this obviate the need for tags? If a review conversation is satisfactorily resolved, then it gets clicked as resolved. Whether the suggestion is "optional" or "required" seems to me to depend on the reviewers vigor which would manifest in his or her persistence in keeping/reopening the conversation.

Reply:
@mrclary I believe the intended benefit of `[optional]` tags are to make clear up front if a request is one that the reviewer would like to see implemented is not a blockers to merging the PR if the contributor is unable to implement it right away, to avoid the user being too intimidated and feeling the reviewer is asking too much of them, while giving reviewers more freedom to suggest a reasonable number of value-added changes with less worry of scaring off the contributor. There have been cases where I've actually wanted something like this, e.g. for additions/improvements to GIFs on the docs, or requesting that they cover a related case, and in the past too for improvements on Spyder code PRs, e.g. adding more detailed tests (beyond the minimum) or refactoring something.

I actually don't disagree with adding (only) `[optional]` tags, as they lack most of the aforementioned downsides of `[required]` tags, are inherently scoped to only be needed/desired in a limited number of specific cases, and would be a more a tool to enable clear communication between the reviewer and author rather than a bureaucratic requirement.

The problem, evidently, with relying on the "reviewers vigor" is that, for very _vigorous_ reviewers like myself who tend to review PRs very thoroughly, it can evidently be too intimidating for some newer contributors (even in the case of all review comments being made as one-click suggestions that don't require any conversation or manual resolution). The purpose of this PR is to limit and codify the review process to avoid these situations in the future. While its unfortunate that a relatively small minority of cases requires an additional burden of restrictions to be applied to all reviewers acting in good faith, given the outsize harm caused by those incidents to all involved, it will hopefully at least be a net nuetral long term, particularly taking an approach that balances the needs of both authors and reviewers.

Reply:
I think if there's not interest in labeling feedback, I'm not sure what else these documents/guidelines do. I guess they explicitly state the scope of review that (from my understanding) reviewers are already following? 

And to be clear, that's okay! This is a proposal the team is not required to accept.

Reply:
Hey, sorry if I was unclear. I _agree_ with the making the distinction between optional and required changes, and I _agree_ with the guidance that reviewers should clearly mark optional improvements as `[optional]`. The only part I felt was particularly burdensome and potentially harmful was mandating that all non-optional comments be explicitly labelled as `[required]`, which is also the remaining provision that requires the biggest change to reviewer practice and something I haven't seen successfully used on other projects. @mrclary , does that sufficiently resolve your concerns, or do you feel there are more fundamental issues with the current approach as proposed by this PR?